Modules/System/Permission Sets/src/PermissionImpl.Codeunit.al

```diff
@@ -156,6 +156,22 @@ codeunit 9864 "Permission Impl."
         end;
     end;
 
+    procedure GetObjectionCaptionAndName(var ExpandedPermission: Record "Expanded Permission"; var ObjectCaption: Text; var ObjectName: Text)
+    var
+        AllObj: Record AllObj;
+    begin
+        if ExpandedPermission."Object ID" <> 0 then begin
+            ExpandedPermission.CalcFields("Object Name");
+            ObjectCaption := ExpandedPermission."Object Name";
```

Comment:
I'm a little bit confused by this code.
You're assigne the "Object Name" to the "Object Caption"

Reply:
I think this is normal question as I was also confused. It is wrong on other places also.
In this Codeunit there three such functions. The one I created is the third one but all three of them do assign a value from a FlowField called "Object Name" to a ObjectCaption variable. The background is that in all of those three tables the fields names are misleading:
- "Tenant Permission"
- "Metadata Permission"
- "Expanded Permission"

The goal of this pull request is to align the output on all pages. I am attaching the current state as I think you are from Germany and you can instantly spot the inconsistence.

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/42636293/236445691-886af6a7-7846-4359-b42a-f8c7fcaca8f8.png">
<img width="974" alt="image" src="https://user-images.githubusercontent.com/42636293/236445951-75b75334-7a75-4bbd-b20b-d179719261bd.png">
<img width="602" alt="image" src="https://user-images.githubusercontent.com/42636293/236445815-4d7babed-1789-47e1-b1ec-7854e0caa8cf.png">

If it happens that there is a better idea how to unify the UI, please let me know. I would be happy to seek for the best possible output.

Reply:
While you are there, I would ask you to check one more thing in regard to the page "All Objects with Caption" as it seems it contain further inconsistencies.
I have a W1 SaaS tenant and I open this page using three different languages and then filtering on pages:

1. english
<img width="520" alt="image" src="https://github.com/microsoft/ALAppExtensions/assets/42636293/a5fd556e-38b5-4a84-a9c9-2f4e549dcd72">

2. german (german language app by Microsoft installed)
<img width="490" alt="image" src="https://github.com/microsoft/ALAppExtensions/assets/42636293/fd9d1d62-8547-40fb-8d12-d4a4e0904a20">


3. serbian (serbian language app by third party installed)
<img width="547" alt="image" src="https://github.com/microsoft/ALAppExtensions/assets/42636293/ec0754d7-9a62-4c65-b1c7-f1811be7162a">



The serbian app has all of the page's captions translated but for some reason this page shows the English caption.
I am not sure why this happens for serbian app but not for the german one. The issue is limited to pages only as the table, reports and other types of objects do appear translated properly in this page.

Example - Company Information Page properly translated to serbian when opened:
<img width="511" alt="image" src="https://github.com/microsoft/ALAppExtensions/assets/42636293/c92ddd89-2bdf-4a19-b405-1ee24f927398">

Example with TableData properly properly translated on serbian.
<img width="442" alt="image" src="https://github.com/microsoft/ALAppExtensions/assets/42636293/0d7ad1cc-c6e7-44d7-80fe-4858cfa8c271">


This issue does not have a huge impact but I would like to know the root cause even if it would mean that third party translations do not support page language specific captions in this page.

Reply:
To not cause any breaking changes the existing field will stay as it is for now.

Thus, I think the solution proposed here is good - and hopefully we can get around to cleaning up the confusing "Object Name" field in the permission tables at some point.

As for the page translation issue in Object With Caption that you mentioned above I'm not sure. Can you create a new issue on this and tag me? I'll try to get it assigned to someone who can look into this. 

Resolving this conversation.